internal/validations/clusterlogforwarder/inputs/validate.go

```diff
@@ -49,21 +49,17 @@ func Verify(inputs []loggingv1.InputSpec, status *loggingv1.ClusterLogForwarderS
 		case !validAudit(input, status, extras):
 		case input.Receiver != nil && !extras[constants.VectorName]:
 			badInput("ReceiverSpecs are only supported for the vector log collector")
-		case input.Receiver != nil && input.Receiver.ReceiverTypeSpec == nil:
-			badInput("invalid ReceiverTypeSpec specified for receiver")
 		case input.Receiver != nil && input.Receiver.Type != loggingv1.ReceiverTypeHttp && input.Receiver.Type != loggingv1.ReceiverTypeSyslog:
 			badInput("invalid Type specified for receiver")
-		case input.Receiver != nil && input.Receiver.Type == loggingv1.ReceiverTypeHttp && input.Receiver.Syslog != nil:
+		case input.Receiver != nil && input.Receiver.Type == loggingv1.ReceiverTypeHttp && input.Receiver.ReceiverTypeSpec != nil && input.Receiver.Syslog != nil:
 			badInput("mismatched Type specified for receiver, specified HTTP and have Syslog")
-		case input.Receiver != nil && input.Receiver.Type == loggingv1.ReceiverTypeSyslog && input.Receiver.HTTP != nil:
+		case isReceiverSpecDefined(input) && input.Receiver != nil && input.Receiver.Type == loggingv1.ReceiverTypeSyslog && input.Receiver.ReceiverTypeSpec != nil && input.Receiver.HTTP != nil:
```

Comment:
You are already checking if `input.Receiver != nil && .ReceiverTypeSpec != nil` with `isReceiverSpecDefined()`. Can also use the defined `isSyslogReceiver()`
```suggestion
		case isReceiverSpecDefined(input) && input.Receiver.IsSyslogReceiver() && input.Receiver.HTTP != nil:
```